defi-pulse/src/adapter.ts

```diff
@@ -0,0 +1,74 @@
+import { Requester, Validator } from '@chainlink/external-adapter'
+import { Execute } from '@chainlink/types'
+import { utils } from 'ethers'
+import { getSymbol } from './symbols'
+import { getAllocations } from './index-allocations'
+import { getPriceAdapter } from './priceAdapter'
+import Decimal from 'decimal.js'
+
+export type Index = IndexAsset[]
+
+export type IndexAsset = {
+  asset: string
+  units: Decimal
+  weight: number
+  price: number
```

Comment:
Price should be `undefined`, not set to 0 as default.
```suggestion
  price?: number
```